### DISCUSSION TITLE
Food Data Dictionary - Nutrient Gaps

### DISCUSSION THREAD

**Discussion by LuciaSegovia** [micronutrientsupport/database-architecture#116] [General]
Hi @rbroth @LouiseAnder @kmtang, 

There is one thing that I would like to discuss with you **how to deal with nutrient gaps in the FCT**. 

Our first thougths were to use a hierarchy by geographic location, but there will be cases when a similar food item from the actual location it will be a better fit.

For example: we don't have values for Se in refined maize flour in Malawi FCT. Then, the logical step is to find Se in refined maize flour in Kenya FCT. However, it would be *probably* more accurate to use Se values in refined maize porridge in Malawi than Se for refined maize flour from Kenya. 

So, ideally, we could identify *problematic* food item-nutrient combinations, and flag them. Then, we can find a solution for them. Like, filling the gaps for those food item-nutrient in the original datasets. 

Let me know what you think! And, if that's ok, maybe we should open an issue for that :)

**Comment by rbroth** [comment]
First question: would Malawian refined Maize Flour and Malawian Refined Maize Porridge have the same food genus? I'm assuming not, right?

Currently, all fooditems are problematic in that they have some nutrient values missing, so filling in the nutrient gaps is something we'll have to do. Adding in more data to the original FCTs would be ideal, but I don't see how we can do that without real-world testing of foods in the lab ourselves. So far I've been assuming that (at some point in the future) we'd somehow average the nutrient values for a given food genus in a given country/FCT.

related issue, I think: #49

**Reply by LuciaSegovia** [reply]
> First question: would Malawian refined Maize Flour and Malawian Refined Maize Porridge have the same food genus? I'm assuming not, right?

You're right, they will have different codes. 


> Currently, all fooditems are problematic in that they have some nutrient values missing, so filling in the nutrient gaps is something we'll have to do. Adding in more data to the original FCTs would be ideal, but I don't see how we can do that without real-world testing of foods in the lab ourselves.


We could 'add' data by imputing values from a different data source (i.e Se value from a closely related food item in the same fct, an average of Se values from a paper, etc.) while we wait for the ideal but implausible analytical value to arrive. Of course, we would have to add the appropiate metatdata to those values. So, that's something that we might need to discuss as well. 

> So far I've been assuming that (at some point in the future) we'd somehow average the nutrient values for a given food genus in a given country/FCT.

I don't understand what you meant in here, sorry :(

**Reply by rbroth** [reply]
> Se value from a closely related food item in the same fct, an average of Se values from a paper, etc.

 Yes that sounds sensible. I assume this would have to be done manually? Regarding metadata, do you mean the citation for that fooditem? E.g. You change the Selenium Values for a tinned tomato soup based on a paper, and the citation text turns into: "Maggi tinned tomato soup | cite: Doe et al., 'Malawian Food Comp Table' 2008; Segovia et al. 'MAPS Project' 2021 ? 


### Re: Averaging fooditem values: 

#### fooditem table values for malawian tomato soup:
| original food name | food genus | energy content | Carbohydrate Content | Salt Content
| ----- | ---- | ---- | ----- | ----- |
| Maggi Tinned Tomato Soup | Tomato Soup | 300 kcal | 6g | NULL |
| Tesco Tinned tomato Soup with Basil | Tomato Soup | 310 kcal | NULL | 2g

#### combined values for malawian tomato soup :
 | food genus | energy content | Carbohydrate Content | Salt Content
 | ---- | ---- | ----- | ----- |
| Tomato Soup | 305 kcal  | 6g | 2g

Given a food genus, if there are multiple food items with that food genus for the same country/region, we take the averages of all the nutrients to run the calculations. (Or we may use a more sophisticated algorithm than the mean, that's to be decided)

**Reply by LuciaSegovia** [reply]
> Regarding metadata, do you mean the citation for that fooditem?

I mean that users should be able to 'trace' back where every data entry is coming from, but particularly those that *we* impute from other sources. I am not 100% sure how that would be in term of structure, because it would be almost like a duplicate of a fct but with the references of the values. 



> Re: Averaging fooditem values:

That makes complete sense, at least for now :)
